### PR TITLE
IBP-5540 restrictions on location management

### PR DESCRIPTION
**Pull request by cuenyad**
Please, review.

Included:
https://github.com/IntegratedBreedingPlatform/Workbench/pull/685

### CODE REVIEW

File: src/main/java/org/ibp/api/java/impl/middleware/common/validator/LocationValidator.java [hunk 1]
```diff
@@ -40,6 +41,7 @@ public class LocationValidator {
 	private static final Integer LOCATION_NAME_MAX_LENGTH = 60;
 	private static final Integer LOCATION_ABBR_MAX_LENGTH = 12;
 	private static final Set<String> LOCATIONS_NOT_DELETABLES = new HashSet<>(Arrays.asList("Unspecified Location", "Default Seed Store", "Default Breeding Location"));
+	private static final List<Integer> RESTRICTED_LOCATION_TYPE = Arrays.asList(401, 405, 406);
```
Comment (clarysabel) [inline src/main/java/org/ibp/api/java/impl/middleware/common/validator/LocationValidator.java:44]: @cuenyad make it plural

File: src/main/resources/messages_en.properties [hunk 2]
```diff
@@ -698,6 +698,9 @@ location.name.max.length=The location name exceeds maximum length of {0} charact
 location.is.used.in.field.map=The location Id {0} is used in a Field Map
 location.not.deletable=The location Id {0} can not be deletable because it is used internally in the system.
 location.not.editable=The location Id {0} can not be editable because it is used internally in the system.
+location.with.location.type.restricted.cannot.edited=The location with a type {0} can not be edited.
```
Comment (clarysabel) [inline src/main/resources/messages_en.properties:701]: @cuenyad remova "a" from 3 properties

File: src/main/java/org/ibp/api/java/impl/middleware/common/validator/LocationValidator.java (hunk 1 above)
Comment (cuenyad) [reply src/main/java/org/ibp/api/java/impl/middleware/common/validator/LocationValidator.java]: done.

File: src/main/resources/messages_en.properties (hunk 2 above)
Comment (cuenyad) [reply src/main/resources/messages_en.properties]: done.

**Pull request merged**